presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSourceProvider.java

```diff
@@ -110,42 +111,43 @@ private InternalTable getInternalTable(ConnectorSession connectorSession, Connec
         InformationSchemaTableHandle handle = (InformationSchemaTableHandle) tablehandle;
         Set<QualifiedTablePrefix> prefixes = handle.getPrefixes();
 
-        return getInformationSchemaTable(session, handle.getCatalogName(), handle.getSchemaTableName(), prefixes);
+        return getInformationSchemaTable(session, handle.getCatalogName(), handle.getSchemaTableName(), prefixes, handle.getLimit());
     }
 
-    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Set<QualifiedTablePrefix> prefixes)
+    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Set<QualifiedTablePrefix> prefixes, OptionalLong limit)
     {
         if (table.equals(TABLE_COLUMNS)) {
-            return buildColumns(session, prefixes);
+            return buildColumns(session, prefixes, limit);
         }
         if (table.equals(TABLE_TABLES)) {
-            return buildTables(session, prefixes);
+            return buildTables(session, prefixes, limit);
         }
         if (table.equals(TABLE_VIEWS)) {
-            return buildViews(session, prefixes);
+            return buildViews(session, prefixes, limit);
         }
         if (table.equals(TABLE_SCHEMATA)) {
-            return buildSchemata(session, catalog);
+            return buildSchemata(session, catalog, limit);
         }
         if (table.equals(TABLE_TABLE_PRIVILEGES)) {
-            return buildTablePrivileges(session, prefixes);
+            return buildTablePrivileges(session, prefixes, limit);
         }
         if (table.equals(TABLE_ROLES)) {
-            return buildRoles(session, catalog);
+            return buildRoles(session, catalog, limit);
         }
         if (table.equals(TABLE_APPLICABLE_ROLES)) {
-            return buildApplicableRoles(session, catalog);
+            return buildApplicableRoles(session, catalog, limit);
         }
         if (table.equals(TABLE_ENABLED_ROLES)) {
-            return buildEnabledRoles(session, catalog);
+            return buildEnabledRoles(session, catalog, limit);
         }
 
         throw new IllegalArgumentException(format("table does not exist: %s", table));
     }
 
-    private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> prefixes)
+    private InternalTable buildColumns(Session session, Set<QualifiedTablePrefix> prefixes, OptionalLong limit)
     {
         InternalTable.Builder table = InternalTable.builder(informationSchemaTableColumns(TABLE_COLUMNS));
+        long recordCnt = 0;
```

Comment:
Nit: we don't abbreviate in our code style. Please name this `recordCount` or just `count`.